elasticSnorkel.ipynb

```diff
@@ -0,0 +1,410 @@
+   "source": [
+    "## Intializing Elasitcsearch\n",
+    "\n",
+    "We begin by intializing Elasticsearch with the names of our index and type. You should have Elasticsearch installed and already ran the \"bin/elasticsearch\" command. For more information you can visit: https://www.elastic.co/guide/en/elasticsearch/guide/current/running-elasticsearch.html.\n",
+    "\n",
+    "Elasticsearch uses slightly difference terminology from tradational databases. \n",
+    "\n",
+    "This is the following mapping:\n",
+    "Database = Index\n",
+    "Table = Type\n",
+    "Row = Document\n",
+    "\n",
+    "Here we intialize an object where corpus will be the name of the index with a single type articles."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from snorkel import SnorkelSession\n",
+    "from snorkel.models import Document, Sentence\n",
+    "from elastics import elasticSession ,printResults,deleteIndex\n",
+    "\n",
+    "session = SnorkelSession()\n",
+    "eSearch=elasticSession(\"corpus\",\"articles\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Check Connection\n",
+    "\n",
+    "Before we begin indexing, we can check our connection with Elasticsearch as well as our existing indices. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Index Information: \n",
+      " \n",
+      "health status index uuid pri rep docs.count docs.deleted store.size pri.store.size\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "eSearch.getIndices()\n",
+    "\n",
+    "#To delete an index by name or _all to delete all indices\n",
+    "#deleteIndex(indexName)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Indexing the Corpus\n",
+    "\n",
+    "Now that we have an established connection we can begin indexing. Here we establish what fields each document in our type will contain, a line number, the sentence and the ner_tags converted to a string. Once everything has been indexed we can see its status with the amount of documents it contains as well as the size."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Beginning Indexing\n",
+      "Documents added 67820\n",
+      "Index Information: \n",
+      " \n",
+      "health status index  uuid                   pri rep docs.count docs.deleted store.size pri.store.size\n",
+      "yellow open   corpus VlTi5iDNQXi2cLp9sYAldQ   5   1      67820            0     45.6mb         45.6mb\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "count = 0\n",
+    "print \"Beginning Indexing\"\n",
+    "for p in session.query(Document):\n",
```

Comment:
Here is where we should really both (1) automate and (1) generalize:

First, the goal should be to make this a one liner.  We can assume that the context hierarchy being used is just the default one (`Document` -> `Sentence` -> `Span`) and that the user wants to index their entire set of loaded contexts in the database, to make things simple.

Second, and more important: this code is currently only useful if someone is focused on candidates which are pairs of NER tags, which is only a small amount of Snorkel applications.  The spec we should aim for is to take in an arbitrary set of candidates (again, feel free to assume we want to just use all candidates) and then can just encode these candidates in a similar serialized format, to keep the rest of the code mostly consistent